datafusion/proto/src/to_proto.rs

```diff
@@ -224,10 +224,12 @@ impl From<&DataType> for protobuf::arrow_type::ArrowTypeEnum {
                 fractional: *fractional as u64,
             }),
             DataType::Decimal256(_, _) => {
-                unimplemented!("The Decimal256 data type is not yet supported")
+                unimplemented!("Proto serialization error: The Decimal256 data type is not yet supported")
```

Comment:
If we used `impl TryFrom` instead of `impl From`, we could return errors here. This could be done as a separate Pr.

Reply:
https://github.com/apache/arrow-datafusion/issues/3401